flink-connector-gcp-pubsub/src/main/java/org/apache/flink/streaming/connectors/gcp/pubsub/emulator/EmulatorCredentialsProvider.java

```diff
@@ -20,12 +20,14 @@
 import com.google.api.gax.core.CredentialsProvider;
 import com.google.auth.Credentials;
 
+import java.io.Serializable;
+
 /**
  * A CredentialsProvider that simply provides the right credentials that are to be used for
  * connecting to an emulator. NOTE: The Google provided NoCredentials and NoCredentialsProvider do
  * not behave as expected. See https://github.com/googleapis/gax-java/issues/1148
  */
-public final class EmulatorCredentialsProvider implements CredentialsProvider {
+public final class EmulatorCredentialsProvider implements CredentialsProvider, Serializable {
```

Comment:
I wonder why do we need it

Reply:
Unlike PubSubSinkV1 we use CredentialsProvider as argument instead of Credentials, hence the class itself is passed to the Job and needs to be serialized, previously Credentials were the argument and hence this wasn't needed to be serialized

Reply:
Looks like we missed it
according to Flink contribution guide there should be `serialVersionUID` 
https://flink.apache.org/how-to-contribute/code-style-and-quality-java/#java-reflection

Reply:
Seems like same should be done for `GcpPublisherConfig`

Reply:
Thanks @snuyanzin , Addressed